### PR TITLE
Fixes gas enumeration error and Chat Really error

### PR DESCRIPTION
**Pull request by Bod9001**
threads are weird man, 
also no originator

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs [hunk 1]
```diff
@@ -147,39 +147,44 @@ bool DistanceCheck(Vector3 playerPos)
...
-			var npcs = Physics2D.OverlapCircleAll(chatEvent.originator.AssumedWorldPosServer(), 14f, npcMask);
-			foreach (Collider2D coll in npcs)
+
+			if (chatEvent.originator != null)
 			{
-				var npcPosition = coll.gameObject.AssumedWorldPosServer();
-				if (MatrixManager.Linecast(chatEvent.position,LayerTypeSelection.Walls,
-					 layerMask,npcPosition).ItHit ==false)
+				//Get NPCs in vicinity
+				var npcs = Physics2D.OverlapCircleAll(chatEvent.position, 14f, npcMask);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs:154]: You've removed the `chatEvent.originator.AssumedWorldPosServer()` so you dont need to null check anymore

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs:154]: the problem is poly was picking up action text

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs:154]: so I don't know if it's expecting, all the chat events to be players

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs:154]: didn't someone separate messages into channels so we don't mix up action and voice?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs:154]: Polly still picks it up if I pass it

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs:154]: ok, minor issue then considering this might fix the server freezing

**Review by corp-0 — APPROVED** [review]

**Review by ThatDan123 — APPROVED** [review]

**Pull request merged**